### PR TITLE
Additional map interactions

### PR DESCRIPTION
**Pull request by emaohn**
- Zoom to marker when travel object is clicked in itinerary
- Draw paths only between events under selected date in itinerary
- Zoom in/out to encompass all of selected days' events when toggling between days in itinerary

### CODE REVIEW

File: step-capstone/src/components/Map.js [hunk 1]
```diff
@@ -108,14 +128,15 @@ class MapComponent extends React.Component {
     // construct hashmap with key: travelObject id, value: marker object
     return list.reduce((hashMap, item) => {
```
Comment (hieudan225) [inline step-capstone/src/components/Map.js:129]: maybe, we could name the hashMap as `travelObject2Marker`?

File: step-capstone/src/components/Map.js [hunk 2]
```diff
@@ -108,14 +128,15 @@ class MapComponent extends React.Component {
...
       if (item.type !== "flight") {
-        hashMap.set(item.id, { marker: this.addMarker(item.coordinates, item.type, item.id), type: item.type });
+        hashMap.set(item.id, { marker: this.addMarker(item.coordinates, item.type), type: item.type });
```
Comment (hieudan225) [inline step-capstone/src/components/Map.js:131]: maybe, we could name the hashMap `travelObject2Marker`?

File: step-capstone/src/components/Trip.js [hunk 3]
```diff
@@ -21,35 +21,45 @@ export default class Trip extends React.Component {
                 endDate: new Date(),
                 destinations: "",
                 description: ""
+            },
+            loaded: false,
+            selectedObject: null,
+            today: {
```
Comment (hieudan225) [inline step-capstone/src/components/Trip.js:27]: maybe consider naming `today` as `displayDate` for consistency

Comment (emaohn) [reply step-capstone/src/components/Trip.js:27]: good suggestion! will do :)

**Pull request merged**